src/LinearSolvers/LinearSolvers_kernels.jl

```diff
@@ -0,0 +1,20 @@
+using Requires
+@init @require CUDAnative = "be33ccc6-a3ff-5ff2-a52e-74243cff1e17" begin
+  using .CUDAnative
+end
+
+function linearcombination!(Q, cs, Xs, increment::Bool)
```

Comment:
This looks like a mat-vec multiply: is doing this via vectors of tuples really more efficient?

Reply:
It is a mat-vec multiply, but it is special in a way. The size of the matrix is m x length(state_vector) where m <= 10. Not sure if viewing this operation as a mat-vec
multiply is the best way to reason about its performance. Anyway, this kernel is probably
not yet optimal, but it is faster that what was in there before (a deep broadcasted expression).